Mathlib/NumberTheory/Fermat.lean

```diff
@@ -61,6 +61,11 @@ theorem fermatNumber_eq_prod_add_two (n : ℕ) :
 theorem fermatNumber_succ (n : ℕ) : fermatNumber (n + 1) = (fermatNumber n - 1) ^ 2 + 1 := by
   rw [fermatNumber, pow_succ, mul_comm, Nat.pow_mul']
   rfl
+theorem fermat_succ_succ (n : ℕ) :
```

Comment:
Please restore the blank line separating this declaration from the previous. Also, as I already suggested, this theorem needs a different name; it's not as if anyone who encounters "fermatNumber (n + 2)" in their goal state will automatically be wanting to use this theorem, it's more specialised than that. I think it should have a name based on the RHS of the statement as well. Perhaps "fermatNumber_eq_fermatNumber_sq_sub_two_mul_fermatNumber_sub_one_sq".

Reply:
I renamed as suggested.
Also I was thinking a bit about if we rather want the int version of this, but I am not really certain. Perhaps it would be a good idea having the inequality that shows that the Nat subtraction will always yield the same result as the Int subtraction (after casting)?
See added lemma

Reply:
I’d prefer reformulating to use Int subtraction. Nat subtraction is a fundamentally un-mathematical operation.